### PR TITLE
[FEAT] TCP Support in Graylog connection establishment

### PR DESCRIPTION
**Pull request by carlosljr**
With GoGelf v2, it supports both TCP and UDP protocols. Our lib will choose the correct gelfWriter based on the chosen transport protocol.

### CODE REVIEW

File: glbgelf.go [hunk 1]
```diff
@@ -21,11 +20,12 @@ var (
 
 // Gelf struct
 type Gelf struct {
-	writer      *gelf.Writer
+	writer      interface{}
```
[the remark this answers is not in the document]
Comment (carlosljr) [reply glbgelf.go]: Yeah, both return `GelfWriter` struct. However, this struct hasn't `WriteMessage` implemented. I call this directly from `TCPWriter` or `UDPWriter`. I think the `WriteMessage` called from `GelfWriter` is a working in progress. If you see, you have only a `Close()` function implemented for `GelfWriter` struct.

File: glbgelf.go [hunk 2]
```diff
@@ -101,13 +101,37 @@ func (g *Gelf) SendLog(extra map[string]interface{}, loglevel string, messages .
...
+		return g.writer.(*gelf.TCPWriter).WriteMessage(m)
+	} else {
+		gUDPWriter := g.writer.(*gelf.UDPWriter)
+		gUDPWriter.CompressionType = 2
+		return gUDPWriter.WriteMessage(m)
+	}
+}
+
+// It will return the correct gelfWriter based on the chosen transport protocol
+func GetWriter(protocol, graylogAddr string) (interface{}, error) {
```
[the remark this answers is not in the document]
Comment (carlosljr) [reply glbgelf.go]: Same as above.

I agree with you. Calling `GelfWriter` would be more indicated. However, this doesn't have the function that a I need for.

File: glbgelf.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nettoclaudio) [reply glbgelf.go]: Hmm... My bad, I mean `gelf.Writer` interface.

[the remark this answers is not in the document]
Comment (carlosljr) [reply glbgelf.go]: Thanks, @nettoclaudio !

The current change was made [in this commit](https://github.com/globocom/glbgelf/commit/ba02cdaf8df93727e7d75706287891517f14ab5e)!

[the remark this answers is not in the document]
Comment (nettoclaudio) [reply glbgelf.go]: Nice!

**Pull request merged**